ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -338,8 +338,14 @@ let assert_gpus_available ~__context ~self ~host =
   let reqd_groups =
     List.map (fun self -> Db.VGPU.get_GPU_group ~__context ~self) vgpus in
   let is_pgpu_available pgpu vgpu =
-    try Xapi_pgpu.assert_can_run_VGPU ~__context ~self:pgpu ~vgpu; true
-    with _ -> false
+    try
+      Xapi_pgpu.assert_can_run_VGPU ~__context ~self:pgpu ~vgpu;
+      Xapi_gpumon.Nvidia.assert_vgpu_pgpu_are_compatible
```

Comment:
I think we need to make sure that we are calling this only for resume operations and that the metadata is in the db. I'd like to know from @robhoes if assert_gpus_available (called by assert_can_boot_here) is the right place, or where it should be called instead. I remember for the migration check, we had decided not to do it in such assert for a reason (but I don't remember the reason)

Reply:
I have addressed the comments about factoring out stringify and reporting the pGPU in case no metadata can be found.

The problem, from where actually invoking the compatibility check is not yet solved.

Reply:
Most important is that this new check is only done for resuming indeed, because assert_can_boot_here is also called for starting VMs and other operations. If we make sure that the VGPU metadata field is always empty if the VM is not suspended, and that this check then passes, then it is probably fine.